test/URMCompoundManager.invariant.t.sol

```diff
@@ -0,0 +1,183 @@
+      revert("Cancelled proposal was cancelled - invariant violation");
+    } catch {
+      // Expected behavior - cancellation of cancelled proposals should revert
+    }
+  }
+
+  // Cancelled proposals are not executable
+  function invariant_cancelledRollbackIsNotExecutable() public {
+    handler.forEachRollbackByState(ProposalState.Canceled, this._checkCancelledProposalNotExecutable);
+  }
+
+  function _checkCancelledProposalNotExecutable(RollbackProposal memory _proposal) external {
+    try urm.execute(_proposal.targets, _proposal.values, _proposal.calldatas, _proposal.description) {
+      revert("Cancelled proposal was executed - invariant violation");
+    } catch {
+      // Expected behavior - execution of cancelled proposals should revert
+    }
+  }
+
+  function invariant_cannotExecuteRollbackWithoutQueueing() public {
+    handler.forEachRollbackByState(ProposalState.Pending, this._checkPendingProposalNotExecutable);
+  }
+
+  function _checkPendingProposalNotExecutable(RollbackProposal memory _proposal) external {
+    try urm.execute(_proposal.targets, _proposal.values, _proposal.calldatas, _proposal.description) {
+      revert("Pending proposal was executed - invariant violation");
+    } catch {
+      // Expected behavior - execution of pending proposals should revert
+    }
+  }
+
+  function invariant_cannotCancelRollbackWithoutQueueing() public {
+    handler.forEachRollbackByState(ProposalState.Pending, this._checkPendingProposalNotCancelable);
+  }
+
+  function _checkPendingProposalNotCancelable(RollbackProposal memory _proposal) external {
+    try urm.cancel(_proposal.targets, _proposal.values, _proposal.calldatas, _proposal.description) {
+      revert("Pending proposal was cancelled - invariant violation");
+    } catch {
+      // Expected behavior - cancellation of pending proposals should revert
+    }
+  }
+
+  function invariant_cannotExecuteRollbackBeforeTimelockDelay() public {
+    handler.forEachRollbackByState(ProposalState.Queued, this._checkQueuedProposalNotExecutable);
+  }
+
+  function _checkQueuedProposalNotExecutable(RollbackProposal memory _proposal) external {
+    try urm.execute(_proposal.targets, _proposal.values, _proposal.calldatas, _proposal.description) {
+      revert("Queued proposal was executed - invariant violation");
+    } catch {
+      // Expected behavior - execution of queued proposals should revert
+    }
+  }
+
+  function invariant_cannotQueueRollbackAfterQueuableDurationPasses() public {
+    handler.forEachRollbackByState(ProposalState.Expired, this._checkExpiredProposalNotQueueable);
+  }
+
+  function _checkExpiredProposalNotQueueable(RollbackProposal memory _proposal) external {
+    try urm.queue(_proposal.targets, _proposal.values, _proposal.calldatas, _proposal.description) {
+      revert("Expired proposal was queued - invariant violation");
+    } catch {
+      // Expected behavior - queuing of expired proposals should revert
+    }
+  }
```

Comment:
@wildmolasses Is it alright to use try catch in invariant testing ? 
Basically what i'm doing is , fetching all the proposals in the expected state and doing tests on them to expect a revert